packages/ui/components/combobox/src/LionCombobox.js

```diff
@@ -43,6 +43,10 @@ export class LionCombobox extends LocalizeMixin(OverlayMixin(LionListbox)) {
       requireOptionMatch: {
         type: Boolean,
       },
+      allowCustomChoice: {
```

Comment:
I think this leaves us with two properties that have the same goal, but are each others inverse (allowCustomChoice === ! requireOptionMatch). 

We usually have these rule for a boolean flag: 
1. it should be `false` by default, (so the non-default can be enabled with a boolean attribute).
2. it should preferably not be a negation, since usually it's harder to reason about

Why do we have `this.requireOptionMatch = true` in the constructor of combobox, then?
Because in this case:
- we thought allowing a custom value should be the default behavior
- we didn't implement this default yet, bc it would be a breaking change. And we want to do this in the future (so then we would comply to rule 1 above again and in our constructor we would do `this.requireOptionMatch = false` again, and then we could introduce `require-option-match` attribute).

Hope this makes sense to you? :)

If so, could you maybe stick to `requireOptionMatch` everywhere? 
And then it would be good if we explained the above in our code, making clear we want to switch the default behavior of the combobox in. a future release

Reply:
See below how `allowCustomChoice` makes more sense in the broader picture (it just means that "no custom choice" will stay the default of combobox then and all reqs are still met)